src/WebJobs.Extensions.DurableTask/Listener/FunctionExecutionHelper.cs

```diff
@@ -24,42 +25,43 @@ internal static class FunctionExecutionHelper
 
             try
             {
-                bool executedUserCode = false;
-                var triggeredFunctionData = new TriggeredFunctionData()
-                {
-                    TriggerValue = triggerInput.TriggerValue,
-                    ParentId = triggerInput.ParentId,
-                    InvokeHandler = userCodeHandler =>
-                    {
-                        executedUserCode = true;
-                        return triggerInput.InvokeHandler(userCodeHandler);
-                    },
-#if !FUNCTIONS_V1
-                    TriggerDetails = triggerInput.TriggerDetails,
-#endif
-                };
-#pragma warning restore CS0618
+                context.ExecutorCalledBack = false;
```

Comment:
I'm confused as to what `context.ExecutorCalledBack` is used for, as it appears to be the exact same valuethat the local variable `executedUserCode` was measuring before. They both are false until they are set to true immediately before the rest of the invoke handler is called.

From what I can tell, embedding this boolean on the DurableCommonContext just requires more work on the caller of this method.

Reply:
I think my intention was to save an object allocation... instead of creating an extra indirection through a `TriggereredFunctionData` wrapper around the userCodeHandler, I just use a boolean flag on the already existing DurableCommonContext object to check if the executor called back.

The previous version had a bug also, I think, since it created this `TriggeredFunctionData` wrapper but then doesn't pass it on as a parameter to the `executor.TryExecuteAsync`.

Reply:
Good catch on that bug. I wonder what doofus wrote the original code here...

I definitely understand your point about removing an object allocation. My concern is that it then puts a burden on the `InvokeHandler` implementations in the respective middleware to set `context.ExecutorCalledBack` to `true`, which would be easy to forget.

That being said, I doubt we will be touching the existing InvokeHandlers too much, and I don't anticipate us adding any new orchestration middleware in the near future, so I could go either way on the issue.